servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TaskBasedAsyncPublisherOperator.java

```diff
@@ -56,20 +57,34 @@ public String toString() {
         }
     };
 
+    private volatile boolean hasOffloaded;
```

Comment:
inline with comment above ... lets move this into the implementations to clarify expected thread access pattern and avoid shared state between subscribe/subscription when not necessary.